main.py

```diff
@@ -140,23 +157,40 @@ def main():
         print(f"Selected Tasks: {task_names}")
 
     results = {}
-    if args.generations_path:
+    if args.generations_path and os.path.exists(args.generations_path):
+        if args.generation_only:
+            print("generation mode only, skip the evaluation")
+            return
         # here we don't generate code but only evaluate previously computed generations
+        if args.generation_only:
+            return
         if accelerator.is_main_process:
             print("evaluation only mode")
         evaluator = Evaluator(accelerator, None, None, args)
         for task in task_names:
             results[task] = evaluator.evaluate(task)
-
     else:
         # here we generate code and save it (evaluation is optional but True by default)
         print("Loading the model and tokenizer")
-        model = AutoModelForCausalLM.from_pretrained(
-            args.model,
-            revision=args.revision,
-            trust_remote_code=args.trust_remote_code,
-            use_auth_token=args.use_auth_token,
-        )
+        if args.max_memory_per_gpu:
+            import torch
+            model = AutoModelForCausalLM.from_pretrained(
+                args.model,     
+                revision=args.revision,
+                trust_remote_code=args.trust_remote_code,
+                use_auth_token=args.use_auth_token,                   
+                device_map="auto", 
+                torch_dtype=torch.bfloat16,
```

Comment:
this [PR](https://github.com/bigcode-project/bigcode-evaluation-harness/pull/61) adds a precision flag that we could use to choose precision here. (btw if you have ~40GB you might not need offloading for the 15b model, there was an issue that took unnecessary memory and was fixed in that PR)